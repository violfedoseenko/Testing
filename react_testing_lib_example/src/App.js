// import { useState, useEffect } from "react";
// // import Users from "./users/users";

// const App = () => {

//   const [data, setData] = useState(null)
//   const [toggle, setToggle] = useState(false)
//   const [value, setValue] = useState('')

//   const onClick = () => setToggle(prev => !prev)

//   useEffect(() => {
//     setTimeout(() => {
//       setData({})
//     }, 1000)
//   }, [])

//   return (
//     <>
//       <h1 data-testid="value-elem">{value}</h1>
//       {toggle === true && <div data-testid="toggle-elem">toggle</div> }
//       {data && <div style={{color: "red"}}>data</div>}
//       <div>hello</div>
//       <button data-testid="toggle-btn" onClick={onClick}>click me</button>
//       <input onChange = {e => setValue(e.target.value)} type="text" placeholder="input value..."/>
//       {/* <Users /> */}
//     </>
//   )

// }

// export default App;


import NavBar from './components/NavBar/NavBar';
import AppRouter from './router/AppRouter';

const App = () => {
	return ( 
		<>
      <NavBar/>
      <AppRouter />
		</>
	 );
}
 
export default App;