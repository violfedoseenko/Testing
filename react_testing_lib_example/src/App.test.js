import { render, screen, fireEvent, userEvent } from '@testing-library/react';
import App from './App';

describe("Test App", () => {
	test('1', () => {
		render(<App />);
		const helloElement = screen.getByText(/hello/i);
		const btn = screen.getByRole('button');
		const input = screen.getByPlaceholderText(/input value/i);
		expect(helloElement).toBeInTheDocument();
		expect(btn).toBeInTheDocument();
		expect(input).toBeInTheDocument();
		expect(input).toMatchSnapshot();
	});

	test('2', async () => {
		render(<App />);
		// const helloElement = screen.queryByText(/hello2/i);
		// expect(helloElement).toBeNull();

		//eslint-disable-next-line
		screen.debug();
		const dataElement = await screen.findByText('data');
		expect(dataElement).toBeInTheDocument();
		expect(dataElement).toHaveStyle({color: 'red'});
		//eslint-disable-next-line
		screen.debug();

	});

	test('click event', async () => {
		render(<App />);
		const btn = screen.getByTestId('toggle-btn')
		// тут лучше использовать query так как изначально этого элемента на странице нет
		// и надо проверить что он отстутствует

		expect(screen.queryByTestId('toggle-elem')).toBeNull()
		fireEvent.click(btn)
		//eslint-disable-next-line
		expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument()
		fireEvent.click(btn)
		expect(screen.queryByTestId('toggle-elem')).toBeNull()
	});

	test('input event via fireEvent', async () => {
		render(<App />);
		const input = screen.getByPlaceholderText(/input value/i);
		expect(screen.queryByTestId('value-elem')).toContainHTML('')

		// fireEvent - искуственное событие
		fireEvent.input(input, {
			target:{
				value: '123123'
			}
		})
		expect(screen.queryByTestId('value-elem')).toContainHTML('123123')
	});

	test('input event via userEvent', async () => {
		render(<App />);
		const input = screen.getByPlaceholderText(/input value/i);
		expect(screen.queryByTestId('value-elem')).toContainHTML('')

		// userEvent не работает с конкретными событиями , а воспроизводит действия пользователя 
		// (движения мыши, нажатия кнопок, etc..)
		userEvent.type(input, '123123')

		expect(screen.queryByTestId('value-elem')).toContainHTML('123123')
	});


})







