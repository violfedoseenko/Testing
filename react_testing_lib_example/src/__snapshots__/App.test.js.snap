// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App 1 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`Test App renders learn react 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`Test App renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`Test App renders learn react link 2`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
